File: app/src/main/java/com/ladsoft/bakingapp/mvp/RecipeMvp.java
package com.ladsoft.bakingapp.mvp;

import com.ladsoft.bakingapp.entity.Recipe;

import java.util.HashMap;

public interface RecipeMvp {
    interface View {
        void showRefresh(boolean show);
        void onRecipeLoaded(Recipe recipe);
        void onRecipeLoadError();
    }

    interface Model {
        void setListener(Callback listener);
        void detachListener();
        void loadRecipe(long recipeId);

        interface Callback {
            void onDataLoaded(Recipe recipe);
            void onDataLoadError();
        }
    }

    interface StateContainer {
        String EXTRA_RECIPE_ID = "extra_recipe_id";

        long getRecipeId();

        HashMap<String, Object> getStateMap();
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/mvp/RecipeMvp.kt
package com.ladsoft.bakingapp.mvp

import com.ladsoft.bakingapp.entity.Recipe

interface RecipeMvp {
    interface View {
        fun showRefresh(show : Boolean)
        fun onRecipeLoaded(recipe: Recipe)
        fun onRecipeLoadError()
    }

    interface Model {
        fun setListener(listener: Callback)
        fun detachListener()
        fun loadRecipe(recipeId: Long)

        interface Callback {
            fun onDataLoaded(recipe: Recipe)
            fun onDataLoadError()
        }
    }

    interface StateContainer {
        val EXTRA_RECIPE_ID: String get() = "extra_recipe_id"

        var recipeId : Long

        fun getStateMap() : HashMap<String, Any>
    }
}